### PR TITLE
added basic instructions on BYOC

### PR DESCRIPTION
**Pull request by mxplusb**
Added basic instructions for how to generate your own root or intermediate CAs based off information in k3s#1868

### For Rancher (product) docs only
When contributing to docs, please update the versioned docs. For example, the docs in the v2.6 folder of the `rancher` folder.

Doc versions older than the latest minor version should only be updated to fix inaccuracies or make minor updates as necessary. The majority of new content should be added to the folder for the latest minor version.

### CODE REVIEW

**Review by btat — DISMISSED** [review]

**Comment by mxplusb** [comment]
@btat updated!

File: content/k3s/latest/en/advanced/_index.md [hunk 1]
```diff
@@ -458,3 +459,68 @@ K3S_TOKEN: Defines the key required by the server to offer the HTTP config resou
...
+openssl req -x509 -new -nodes -key server-ca.key -sha256 -days 3560 -out server-ca.crt -addext keyUsage=critical,digitalSignature,keyEncipherment,keyCertSign -subj '/CN=k3s-server-ca'
+openssl req -x509 -new -nodes -key request-header-ca.key -sha256 -days 3560 -out request-header-ca.crt -addext keyUsage=critical,digitalSignature,keyEncipherment,keyCertSign -subj '/CN=k3s-request-header-ca'
+```
+
+## HashiCorp Vault
+
+This example uses an existing self-signed PKI infrastructure in [HashiCorp Vault](https://www.vaultproject.io/) to generate intermediate CAs.
+
+```
+export ROOT_DOMAIN="you-domain-here.com"
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:487]: Use a RFC 2606 reserved domain for documentation:
```suggestion
export ROOT_DOMAIN="example.com"
```

File: content/k3s/latest/en/advanced/_index.md [hunk 2]
```diff
@@ -458,3 +459,68 @@ K3S_TOKEN: Defines the key required by the server to offer the HTTP config resou
 K3S_AGENT_TOKEN: Optional. Defines the key required by the server to offer HTTP config resources to the agents. If not defined, agents will require K3S_TOKEN. Defining K3S_AGENT_TOKEN is encouraged to avoid agents having to know K3S_TOKEN, which is also used to encrypt data.
 
 If no K3S_TOKEN is defined, the first K3s server will generate a random one. The result is part of the content in `/var/lib/rancher/k3s/server/token`. For example, `K1070878408e06a827960208f84ed18b65fa10f27864e71a57d9e053c4caff8504b::server:df54383b5659b9280aa1e73e60ef78fc`, where `df54383b5659b9280aa1e73e60ef78fc` is the K3S_TOKEN.
+
+# Bring Your Own Intermediate CAs
+
+If you want to handle your own TLS chains, you can provide your own root or intermediate CA certificates to replace the default CAs which K3s generates on installation. You can find more discussions and references in [k3s#1868](https://github.com/k3s-io/k3s/issues/1868). It is important to be aware that you must perform these steps before installing K3s or it will not work.
```
Comment (brandond) [inline content/k3s/latest/en/advanced/_index.md:465]:
```suggestion

> It is important to perform these steps on the first server in the cluster before starting K3s for the first time. Making changes to cluster Root Certificate Authority certificates after the cluster has been brought up may render it unusable.

If you want to use custom Certificate Authority data, you can provide your own root or intermediate CA certificates in place of the default CAs which K3s generates on installation. You can find more discussions and references in [k3s#1868](https://github.com/k3s-io/k3s/issues/1868). 
```

**Review by brandond — CHANGES_REQUESTED** [review]
Thank you for the contribution! 

I suggested a couple changes to the wording to highlight the fact that these steps should only be done before bringing up the cluster.

An additional two CAs are required for use with embedded etcd; would you care to include those in this as well?

**Comment by mxplusb** [comment]
> Thank you for the contribution!
> 
> I suggested a couple changes to the wording to highlight the fact that these steps should only be done before bringing up the cluster.
> 
> An additional two CAs are required for use with embedded etcd; would you care to include those in this as well?

I included those, thanks!

**Comment by brandond** [comment]
>>An additional two CAs are required for use with embedded etcd; would you care to include those in this as well?
> I included those, thanks!

I'm not seeing the etcd CAs mentioned, did those changes get pushed?

**Comment by jtravee** [comment]
Hi @mxplusb, please close this PR and create a new one at https://github.com/k3s-io/docs instead. Thanks!

**Comment by mxplusb** [comment]
I wont be making further improvements to the PR as I've run out of time

**Pull request closed without merging**